### ISSUE TITLE
Question about the setting

### ISSUE THREAD

**Issue by zhangyikaii** [zhukaii/SPPR#3]
您好! 或许代码中的Few-Shot Class-Incremental Setting与论文中(或是与常用的benchmark, Tao等人提出的)不相符:

在第二个session及以后, 每个session的训练集中仅能包含当前新增类别的数据, 即保证所有session训练集的标签空间不相交, 但是您的代码中[`my_dataset.py`](https://github.com/zhukaii/SPPR/blob/main/utils/data/my_dataset.py#L53)似乎把第二个session及以后的训练数据累加在一起考虑, 这将导致[训练时](https://github.com/zhukaii/SPPR/blob/main/utils/Trainer.py#L137)进入额外的数据.

因为没有非常细致地看您的代码, 所以上述也可能是错的, 打搅您了.

非常期望您能给出解答, 谢谢!

Hi! Maybe the Few-Shot Class-Incremental Setting in the code is not consistent with the paper (or with the commonly used benchmark, as proposed by Tao _et al._):

In the second session and beyond, each session's training set can only contain data from the currently added class, _i.e._, the label spaces of all sessions are guaranteed to be disjoint, but your code [`my_dataset.py`](https://github.com/zhukaii/SPPR/blob/main/utils/data/my_dataset.py#L53) seems to consider the second session and subsequent training data cumulatively, which will cause additional data input during [training](https://github.com/zhukaii/SPPR/blob/main/utils/Trainer.py#L137).

Since I haven't looked at your code very carefully, the above may be wrong as well, sorry to bother you.

Looking forward to your answer, thanks!

**Comment by zhukaii** [comment]
As stated in our paper, there is no real training stage after the second session(L. 121-142 in Trainer.py). For the convenience of updating the prototypes, we directly involve the second session and subsequent training data in my_dataset.py. You can also save the previous prototypes and involve only the training data in the current session. It's the same.

**Comment by zhangyikaii** [comment]
Thanks for your reply.

Your answer is reasonable because the backbone is indeed not updated after the second session. A more rigorous approach is to save the previous prototypes, because of the random sampling, the code will still result in different prototypes for different sessions.

Thanks again. You have a great approach, and please close the issue if there are no replies.
